scripts/setenv.sh

```diff
@@ -47,7 +47,7 @@ case "${unameOut}" in
       export RPATH=$LD_LIBRARY_PATH;;
     Darwin*)
       # detect llvm version by valid list
-      for llvm_version in 17 16 15 14 NOT_FOUND ; do
+      for llvm_version in 16 15 14 NOT_FOUND ; do
```

Comment:
Why was 17 deleted?

Reply:
scripts will exit if llvm 17 is not installed in the older MacOS version

Reply:
This shouldn't be the case. The code checker for macOS does not use LLVM 17, it still be able to compile.